doc-changelog/action.yml

```diff
@@ -0,0 +1,138 @@
+      Title of the pull request
+    required: true
+    type: string
+
+  pr-number:
+    description: >
+      The pull request number
+    required: true
+    type: string
+
+  token:
+    description: >
+      GitHub token for doing a git commit & push
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.pr-branch }}
+
+    - name: "Set up Python ${{ inputs.python-version }}"
+      uses: ansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: ${{ inputs.use-python-cache }}
+
+    - name: "Install towncrier"
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip towncrier
+
+    - name: "Get pull request type"
+      shell: bash
+      run: |
+        # Check inputs.pr-title is surrounded by double quotes
+        title='${{ inputs.pr-title }}'
+        if [[ "$title" =~ ^\".*\"$ ]]; then
+          echo "PR title is wrapped in double quotes"
+        else
+          echo "PR title is not wrapped in double quotes"
+          title="${{ inputs.pr-title }}"
+        fi
+
+        # Save title into env variable
+        echo "PR-TITLE=$title" >> $GITHUB_ENV
+
+        # Check if pr-title string starts with Add, Remove, Change, Fix, or Bump
+        added=`if [[ $title =~ ^[a,A]dd ]]; then echo "True"; fi`
+        removed=`if [[ $title =~ ^[r,R]emove ]]; then echo "True"; fi`
+        changed=`if [[ $title =~ ^[c,C]hange ]]; then echo "True"; fi`
+        fixed=`if [[ $title =~ ^[f,F]ix ]]; then echo "True"; fi`
+        dependency=`if [[ $title =~ ^[b,B]ump ]]; then echo "True"; fi`
+
+        # If string is not null, then save the pull request type as an env variable
+        if [ ! -z "$added" ]; then
+          echo "PR-TYPE=added" >> $GITHUB_ENV
+        elif [ ! -z "$removed" ]; then
+          echo "PR-TYPE=removed" >> $GITHUB_ENV
+        elif [ ! -z "$changed" ]; then
+          echo "PR-TYPE=changed" >> $GITHUB_ENV
+        elif [ ! -z "$fixed" ]; then
+          echo "PR-TYPE=fixed" >> $GITHUB_ENV
+        elif [ ! -z "$dependency" ]; then
+          echo "PR-TYPE=dependencies" >> $GITHUB_ENV
+        else
+          echo "The pull request title ${{ inputs.pr-title }} is invalid. Please ensure it starts with one of the following keywords:"
+          echo  "Add, Remove, Change, Fix, or Bump (for dependencies)."
+          exit 1
+        fi
+
+    - name: "Create and commit towncrier fragment"
+      shell: bash
+      run: |
+        file="${{ inputs.pr-number }}.${{ env.PR-TYPE }}.md"
+        changelog_file="${{ inputs.path-to-fragment }}/$file"
+
+        # If fragment already exists with ${{ inputs.pr-number }}, remove it and create a new one
+        if [[ "$changelog_file" == *"${{ inputs.pr-number }}."* ]]; then
+          rm ${{ inputs.path-to-fragment }}/${{ inputs.pr-number }}.*
+        fi
```

Comment:
What if I'm doing my fragment in each PR, because maybe I want something more customizable or similar.

I would attach the one that is already in place.

Reply:
I think this is intended for only generating fragments in PRs. I was following these docs for setting up the towncrier command

https://towncrier.readthedocs.io/en/stable/markdown.html

This line `towncrier create -c "Added a cool feature!" 1.added.md` would result in this section in the changelog after doing towncrier build. 

```
### Added

- Added a cool feature! ([#1](https://github.com/twisted/my-project/issues/1))
```

I was taking in the pr-number, the action of the PR in the title like add, fix, etc., and the title of the PR for the phrase that comes before the link to the PR so then it would generate the fragment similar to how it looks when we manually update the changelog (at least in pymechanical).

In the above example, if 1.added.md already existed in the branch, then it would create another file named 1.added.1.md (or something like that), which would then in the end add another line to the changelog with the same PR number. To get around that, I was deleting any fragment that had the same number, so there would only be one fragment per PR number